File: Python/sum-of-good-numbers.py
# Time:  O(n)
# Space: O(1)

# array
class Solution(object):
    def sumOfGoodNumbers(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        return sum(nums[i] for i in xrange(len(nums)) if (i-k < 0 or nums[i-k] < nums[i]) and (i+k >= len(nums) or nums[i+k] < nums[i]))

File: C++/sum-of-good-numbers.cpp
// Time:  O(n)
// Space: O(1)

// array
class Solution {
public:
    int sumOfGoodNumbers(vector<int>& nums, int k) {
        int result = 0;
        for (int i = 0; i < size(nums); ++i) {
            if ((i - k < 0 || nums[i - k] < nums[i]) &&
                (i + k >= size(nums) || nums[i + k] < nums[i])) {
                result += nums[i];
            }
        }
        return result;
    }
};
